tech-specs/src/explore/design/ibc.md

```diff
@@ -0,0 +1,127 @@
+# IBC integration
+
+[IBC](https://arxiv.org/pdf/2006.15918.pdf) allows a ledger to track another ledger's consensus state using a light client. IBC is a protocol to agree the consensus state and to send/receive packets between ledgers.
+
+## Transaction for IBC
+A requester (IBC relayer or user) who wants to execute IBC operations on a ledger sets IBC packet or message like `MsgCreateAnyClient` as transaction data and submit the following transaction. IBC validity predicate is invoked after this transaction execution to check the IBC operation is validated. The trigger to invoke IBC validity predicate is `insert_verifier()` in IBC-related operation to add an address for IBC `InternalAddress::Ibc`.
+
+The transaction is given an IBC packet or message which specifies what to do. Because the packet or message has been encoded and stored in transaction data and there are some types of messages, it has to check what packet or message is given first and then decodes the message or packet. After the process according to the packet or message, it emits an IBC event.
```

Comment:
Hmm not necessarily, the state writes may just happen, right? The tx may need to include extra data (proofs)

Reply:
Thanks, right. I modified that.